### ISSUE TITLE
Uploaded an empty artifact (0 bytes)

### ISSUE THREAD

**Issue by cwbusacker** [saurav0705/check-history-action#56]
Was running this action and the artifact that was uploaded was 0 bytes which made the next workflow get triggered even though no applicable changes were made.
![image](https://github.com/saurav0705/check-history-action/assets/35274973/d08a2e33-83fc-4123-83d1-1fadeb12c258)
![image](https://github.com/saurav0705/check-history-action/assets/35274973/3953c6a9-7d86-4679-a6ce-fc07c434744c)

**Comment by cwbusacker** [comment]
I'm having a difficult time reproducing this problem, but I think it has something to do with:

```
    github.setClient(GIT_TOKEN)
    github.setConfig({
      repo: context.repo.repo ?? '',
      owner: context.repo.owner ?? '',
      issue_number: context.payload.number ?? 0,
      sha: context.payload.after ?? ''
    })
```

I think the `context.payload.after` might be a blank string or null on certain github action triggers, but I can't find any documentation.

I would like to try just `context.sha` and see if that fixes it, but I have no way of verifying since I can't seem to find a reliable way to reproduce. @saurav0705, any suggestions?

**Comment by saurav0705** [comment]
can you download the artifact and see if it's actually empty

**Comment by cwbusacker** [comment]
No. By the time I realize that it was empty, the artifact has been deleted by the workflow itself.  I would assume it is, though, because the logs say it's uploading `0 bytes`.

**Comment by cwbusacker** [comment]
I'm curious if just updating @actions/github will help which is what #66 will do (as well as add a console.error() with the content of the context when the sha is empty)

**Comment by saurav0705** [comment]
@cwbusacker 
i have added a [fix](#67 ) it should be able to fix the issue also it now adds the log for more control

**Comment by cwbusacker** [comment]
I tried that way as well, but the problem is that the head of the pull request may be different after the test is complete. We can’t guarantee the pull request head is the commit sha that that test was run.
